larpandoracontent/LArControlFlow/NeutrinoIdTool.h

```diff
@@ -24,13 +35,201 @@ class NeutrinoIdTool : public SliceIdBaseTool
+    bool GetBestSliceIndex(const MasterAlgorithm *const pAlgorithm, const SliceHypotheses &nuSliceHypotheses, const SliceHypotheses &crSliceHypotheses, unsigned int &bestSliceIndex) const;
+
+    /**
+     *  @brief  Determine if the event passes the selection cuts for training
+     *
+     *  @param  pAlgorithm address of the master algorithm
+     *  @param  purity purity of best slice
+     *  @param  completeness completeness of best slice
+     */
+    bool PassesQualityCuts(const MasterAlgorithm *const pAlgorithm, const float purity, const float completeness) const;
+
+    /**
+     *  @brief  Collect all 2D hits in a supplied list of Pfos and push them on to an existing hit list, check so not to double count
+     *
+     *  @param  pfos input list of pfos
+     *  @param  hitList output list of all 2d hits in the input pfos
+     */
+    void Collect2DHits(const pandora::PfoList &pfos, pandora::CaloHitList &hitList) const;
+
+    /**
+     *  @brief  Count the number of neutrino induced hits in a given list
+     *
+     *  @param  hitList input list of calo hits
+     */
+    unsigned int CountNeutrinoInducedHits(const pandora::CaloHitList &hitList) const;
+
+    /**
+     *  @brief  Use the current MCParticle list to get the nuance code of the neutrino in the event
+     *
+     *  @param  pAlgorithm address of the master algorithm
+     */
+    int GetNuanceCode(const MasterAlgorithm *const pAlgorithm) const;
+
+    /**
+     *  @brief  Select all pfos under the same hypothesis
+     *
+     *  @param  hypotheses the lists of slices under a certain hypothesis
+     *  @param  selectedPfos the list of pfos to populate
+     */
+    void SelectAllPfos(const SliceHypotheses &hypotheses, pandora::PfoList &selectedPfos) const;
+
+    /**
+     *  @brief  Select pfos based on the probability that their slice contains a neutrino interaction
+     *
+     *  @param  nuSliceHypotheses the input neutrino slice hypotheses
+     *  @param  crSliceHypotheses the input cosmic slice hypotheses
+     *  @param  sliceFeaturesVector vector holding the slice features
+     *  @param  selectedPfos the list of pfos to populate
+     */
+    void SelectPfosByProbability(const SliceHypotheses &nuSliceHypotheses, const SliceHypotheses &crSliceHypotheses, const SliceFeaturesVector &sliceFeaturesVector, pandora::PfoList &selectedPfos) const;
+
+    /**
+     *  @brief  Add the given pfos to the selected Pfo list
+     *
+     *  @param  pfos the pfos to select
+     *  @param  selectedPfos the list of pfos to populate
+     */
+    void SelectPfos(const pandora::PfoList &pfos, pandora::PfoList &selectedPfos) const;
+
     pandora::StatusCode ReadSettings(const pandora::TiXmlHandle xmlHandle);
 
-    bool        m_selectAllNeutrinos;               ///< First approach: select all neutrinos, as opposed to selecting all cosmics
-    bool        m_selectOnlyFirstSliceNeutrinos;    ///< First approach: select first slice neutrinos, cosmics for all subsequent slices
```

Comment:
Should probably maintain the old functionality, to be enabled via config. This may be by adding a simple tool (as we use in production right now), or, if you prefer, by maintaining these two Booleans in this tool.

Reply:
Do you mean maintain these as options to bypass this neutrino ID tool? Maybe it makes most sense, as you say, to just keep the simple tool too.